scylla/src/transport/connection.rs

```diff
@@ -1284,7 +1284,7 @@ impl Connection {
                     _ => Err(err.into()),
                 },
                 Response::Result(_) => Ok(query_response.into_query_result()?),
-                _ => Err(ProtocolError::UnexpectedResponse(
+                _ => Err(UserRequestError::UnexpectedResponse(
                     query_response.response.to_response_kind(),
                 )
                 .into()),
```

Comment:
It took me a few seconds to notice the "into()" call, and now I'm more convinced that such conversions are a bad idea. Why not return the variant that we want explicitly instead of relying on conversions?

Reply:
This `into()` call is later removed anyway (commit `batch: return UserRequestError from batch_with_consistency`). But I agree with the suggestion, and will start applying it in the future.